test_workspace_concurrency failing regularly

We've started to see failures in `test_diskutils.py::test_workspace_concurrency` that I'm unable to understand

```
_________________________ test_workspace_concurrency __________________________
Traceback (most recent call last):
  File "C:\projects\distributed\distributed\tests\test_diskutils.py", line 260, in test_workspace_concurrency
    _test_workspace_concurrency(tmpdir, 2.0, 6)
  File "C:\projects\distributed\distributed\tests\test_diskutils.py", line 255, in _test_workspace_concurrency
    assert n_purged >= 0.5 * n_created > 0
AssertionError: assert 0 >= (0.5 * 1200)
```

cc @pitrou , author of that test, in case he's able to explain what is going on

--- a/distributed/distributed-schema.yaml
+++ b/distributed/distributed-schema.yaml
@@ -286,11 +286,10 @@ properties:
               For a list of handlers see the `dask.distributed.Scheduler.handlers` attribute.
 
           multiprocessing-method:
-            type:
-              enum:
-                - spawn
-                - fork
-                - forkserver
+            enum:
+              - spawn
+              - fork
+              - forkserver
             description: |
               How we create new workers, one of "spawn", "forkserver", or "fork"
 

--- a/distributed/distributed-schema.yaml
+++ b/distributed/distributed-schema.yaml
@@ -286,7 +286,11 @@ properties:
               For a list of handlers see the `dask.distributed.Scheduler.handlers` attribute.
 
           multiprocessing-method:
-            type: string
+            type:
+              enum:
+                - spawn
+                - fork
+                - forkserver
             description: |
               How we create new workers, one of "spawn", "forkserver", or "fork"
 

--- a/distributed/tests/test_diskutils.py
+++ b/distributed/tests/test_diskutils.py
@@ -12,6 +12,7 @@ import pytest
 
 import dask
 
+from distributed.compatibility import WINDOWS
 from distributed.diskutils import WorkSpace
 from distributed.metrics import time
 from distributed.utils import mp_context
@@ -214,8 +215,7 @@ def _workspace_concurrency(base_dir, purged_q, err_q, stop_evt, barrier):
 
 
 @pytest.mark.slow
-@pytest.mark.parametrize("timeout,max_procs", [(5.0, 6), (10.0, 16)])
-def test_workspace_concurrency(tmpdir, timeout, max_procs):
+def test_workspace_concurrency(tmpdir):
     """WorkSpace concurrency test. We merely check that no exception or
     deadlock happens.
     """
@@ -227,6 +227,10 @@ def test_workspace_concurrency(tmpdir, timeout, max_procs):
     ws = WorkSpace(base_dir)
     # Make sure purging only happens in the child processes
     ws._purge_leftovers = lambda: None
+
+    # Windows (or at least Windows GitHub CI) has been observed to be exceptionally
+    # slow. Don't stress it too much.
+    max_procs = 2 if WINDOWS else 16
 
     # Run a bunch of child processes that will try to purge concurrently
     barrier = mp_context.Barrier(parties=max_procs + 1)
@@ -244,14 +248,16 @@ def test_workspace_concurrency(tmpdir, timeout, max_procs):
     n_purged = 0
     t1 = time()
     try:
-        while time() - t1 < timeout:
-            # Add a bunch of locks, and simulate forgetting them.
+        # On Linux, you will typically end with n_created > 10.000
+        # On Windows, it can take 60 seconds to create 50 locks!
+        while time() - t1 < 10:
+            # Add a bunch of locks and simulate forgetting them.
             # The concurrent processes should try to purge them.
-            for i in range(50):
+            for _ in range(100):
                 d = ws.new_work_dir(prefix="workspace-concurrency-")
                 d._finalizer.detach()
                 n_created += 1
-            sleep(0.01)
+
     finally:
         stop_evt.set()
         for p in processes:
@@ -270,6 +276,6 @@ def test_workspace_concurrency(tmpdir, timeout, max_procs):
             n_purged += purged_q.get_nowait()
     except queue.Empty:
         pass
-    assert n_created >= 100
+
     # We attempted to purge most directories at some point
     assert n_purged >= 0.5 * n_created > 0

--- a/distributed/utils.py
+++ b/distributed/utils.py
@@ -46,7 +46,7 @@ from dask import istask
 from dask.utils import parse_timedelta as _parse_timedelta
 from dask.widgets import get_template
 
-from .compatibility import PYPY, WINDOWS
+from .compatibility import WINDOWS
 from .metrics import time
 
 try:
@@ -67,11 +67,9 @@ no_default = "__no_default__"
 
 
 def _initialize_mp_context():
-    if WINDOWS or PYPY:
-        return multiprocessing
-    else:
-        method = dask.config.get("distributed.worker.multiprocessing-method")
-        ctx = multiprocessing.get_context(method)
+    method = dask.config.get("distributed.worker.multiprocessing-method")
+    ctx = multiprocessing.get_context(method)
+    if method == "forkserver":
         # Makes the test suite much faster
         preload = ["distributed"]
         if "pkg_resources" in sys.modules:
@@ -87,7 +85,8 @@ def _initialize_mp_context():
             else:
                 preload.append(pkg)
         ctx.set_forkserver_preload(preload)
-        return ctx
+
+    return ctx
 
 
 mp_context = _initialize_mp_context()

